src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertificatesTabCoordinator.swift

```diff
@@ -474,4 +481,86 @@ final class HealthCertificatesTabCoordinator {
 			self.showErrorAlert(title: AppStrings.HealthCertificate.Error.title, error: error)
 		}
 	}
+	
+	private func showPdfGenerationResult(pdfDocument: PDFDocument) {
+		let healthCertificatePDFVersionViewModel = HealthCertificatePDFVersionViewModel(
+			healthCertificate: nil, pdfDocument: pdfDocument
+		)
+		
+		let healthCertificatePDFVersionViewController = HealthCertificatePDFVersionViewController(
+			viewModel: healthCertificatePDFVersionViewModel,
+			onTapPrintPdf: { [weak self] data in
+				self?.showPrintPdf(pdfData: data)
+			},
+			onTapExportPdf: { [weak self] pdfItem in
+				self?.exportPdf(exportItem: pdfItem)
+			}
+		)
+		// The call of showPdfGenerationResult is made possibly in the background while generating the pdfDocument
+		DispatchQueue.main.async { [weak self] in
+			self?.printNavigationController.pushViewController(healthCertificatePDFVersionViewController, animated: true)
+		}
+	}
+	
+	private func showPrintPdf(
+		pdfData: Data
+	) {
+		// swiftlint:disable:next no_plain_print
+		guard UIPrintInteractionController.canPrint(pdfData) else {
+			Log.error("UIPrintInteractionController can't print given pdf data")
+			return
+		}
+
+		DispatchQueue.main.async {
+			let printController = UIPrintInteractionController.shared
+			printController.printingItem = pdfData
+			printController.present(animated: true) { _, success, error in
+				if let error = error {
+					Log.error("Error printing pdf:", error: error)
+				}
+				if !success {
+					Log.info("Failed to print pdf file")
+				} else {
+					Log.info("Did print pdf file successfully")
+				}
+			}
+		}
+	}
+	
+	private func exportPdf(
+		exportItem: PDFExportItem
+	) {
+		let activityViewController = UIActivityViewController(activityItems: [exportItem], applicationActivities: nil)
+		viewController.present(activityViewController, animated: true, completion: nil)
+	}
+	
+	private func showPdfPrintErrorAlert() {
+		let alert = UIAlertController(
+			title: AppStrings.HealthCertificate.PrintPDF.ErrorAlert.pdfGeneration.title,
+			message: AppStrings.HealthCertificate.PrintPDF.ErrorAlert.pdfGeneration.message,
+			preferredStyle: .alert
+		)
+		
+		let faqAction = UIAlertAction(
+			title: AppStrings.HealthCertificate.PrintPDF.ErrorAlert.pdfGeneration.faq,
+			style: .default,
+			handler: { _ in
+				LinkHelper.open(urlString: AppStrings.Links.healthCertificatePrintFAQ)
+			}
+		)
+		faqAction.accessibilityIdentifier = AccessibilityIdentifiers.HealthCertificate.PrintPdf.faqAction
+		alert.addAction(faqAction)
+		
+		let okayAction = UIAlertAction(
+			title: AppStrings.HealthCertificate.PrintPDF.ErrorAlert.pdfGeneration.ok,
+			style: .cancel,
+			handler: { _ in
```

Comment:
weak self

Reply:
Why do we need to capture weak self when we don't need it? if you are referring to alert, it is a variable only available in this function scope.

Reply:
It's just a quick thought to avoid a memory leak. But you've right!